node_modules/oae-core/documentpreview/js/documentpreview.js

```diff
@@ -84,86 +171,85 @@ define(['jquery', 'underscore', 'oae.core', 'lazyload'], function($, _, oae) {
             // Indicates whether or not we attempted to load this page previously
             _isRetryAttempt = _isRetryAttempt || false;
 
-            // Don't reload the page if it has already been loaded
-            if (page.$el) {
+            // Don't reload the page if all of its components have already been loaded
+            if (page.$el && page.$html && page.$text) {
                 return callback();
             }
+
+            // Variables to track progress of HTML and text content loads
+            var htmlDeferred = new $.Deferred();
```

Comment:
Whilst I can see why deferreds are being used here, I don't think we should start to mix them in in ad-hoc way and I'd prefer to take them back out of this PR